lib/ignored-paths.js

```diff
@@ -192,34 +192,24 @@ IgnoredPaths.prototype.contains = function(filepath, category) {
 
 /**
  * Returns a list of dir patterns for glob to ignore
- * @returns {string[]} list of glob ignore patterns
+ * @returns {function()} method to check whether a folder should be ignored by glob
  */
-IgnoredPaths.prototype.getIgnoredFoldersGlobPatterns = function() {
-    var dirs = DEFAULT_IGNORE_DIRS;
-
-    if (this.options.ignore) {
-
-        /* eslint-disable no-underscore-dangle */
-
-        var patterns = this.ig.custom._rules.filter(function(rule) {
-            return rule.negative;
-        }).map(function(rule) {
-            return rule.origin;
-        });
+IgnoredPaths.prototype.getIgnoredFoldersGlobChecker = function() {
+    var dirs = DEFAULT_IGNORE_DIRS.map(function(dir) {
+        return dir + "*";
```

Comment:
so that, `'node_modules/package'` could be reincluded by adding `!node_modules/package` to .eslintignore file.